common/githelper/githelper.go

```diff
@@ -0,0 +1,88 @@
+/*
+ *  Copyright IBM Corporation 2023
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package githelper
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/go-git/go-git/v5"
+)
+
+type gitRemotePath struct {
+	URL            string
+	Branch         string
+	Tag            string
+	Commit         string
+	PathWithinRepo string
+}
+
+// GetGitRemotePathStruct extracts information from the given git path and returns a struct
+func GetGitRemotePathStruct(gitremotepath string) (*gitRemotePath, error) {
+	// reference format - git+ssh://<git remote URL>^<branch,tag,commit>^<path within the repo>
+	// reference format - git+https://<git remote URL>^<branch,tag,commit>^<path within the repo>
+	// reference format - git+ssh://<git remote URL>^branch_<branchname>^<path within the repo>
+	// reference format - git+https://<git remote URL>^branch_<branchname>^<path within the repo>
```

Comment:
Why not reuse existing formats like https://pip.pypa.io/en/stable/topics/vcs-support/?

Reply:
> > Thanks for the PR, Venkat, can you look at my comment and also please update this branch. Thanks.
> 
> Please consider this alternative, adding the feature behind a flag
> 
> ```
> $ move2kube transform --source-repo github.com/myaccount/mysourcerepo.git --cust-repo github.com/myaccount/mycustrepo.git
> ```
> 
> Make it easier to integrate with the UI later and it avoids breaking some assumptions we have made in the CLI, API and UI.

Why would it impact the UI, it would just become an additional feature supported, isn't it? Having a single flag will help avoiding coordinating between the various ways we can get input, isn't it?

Reply:
```
Why would it impact the UI, it would just become an additional feature supported, isn't it? Having a single flag will help avoiding coordinating between the various ways we can get input, isn't it?
```
Would like expose this feature in the UI so that users can provide the URL instead of a zip file for sources and customizations.
If we use a temporary folder it will keep cloning the same repo for every transformation. Making it configurable allows us to clone just once (clone the repo again only if the user tells us to pull some new change).

```
Having a single flag will help avoiding coordinating between the various ways we can get input, isn't it?
```
The format mentioned here https://pip.pypa.io/en/stable/topics/vcs-support/ is also valid.
I suggested using a separate flag for simplicity of implementation. It also makes it less likely to accidentally break the default flows we have right now.